src/main/java/io/github/microcks/testcontainers/MicrocksContainer.java

```diff
@@ -561,6 +564,115 @@ public void downloadAsSecondaryRemoteArtifact(String remoteArtifactUrl) throws A
+
+      return count != null && count.intValue() != 0;
+   }
+
+   /**
+    * Get the invocations' count for a given service, identified by its name and version, for the current invocations' date.
+    *
+    * @param serviceName    mandatory
+    * @param serviceVersion mandatory
+    * @return zero if the given service was not found, or has never been invoked. Else, returns the daily count of invocations for the given service.
+    */
+   public Long getServiceInvocationsCount(String serviceName, String serviceVersion) {
+      return this.doGetServiceInvocationsCount(serviceName, serviceVersion, null);
+   }
+
+   /**
+    * Get the invocations' count for a given service, identified by its name and version, for the given invocations' date.
+    * In unit testing context, it should be useless to pass the invocations date, prefer calling {@link #getServiceInvocationsCount(String, String)}`.
+    *
+    * @param serviceName     mandatory
+    * @param serviceVersion  mandatory
+    * @param invocationsDate nullable
+    * @return zero if the given service was not found, or has never been invoked. Else, returns the daily count of invocations for the given service.
+    */
+   public Long getServiceInvocationsCount(String serviceName, String serviceVersion, Date invocationsDate) {
+      return this.doGetServiceInvocationsCount(serviceName, serviceVersion, invocationsDate);
+   }
+
+   private Long doGetServiceInvocationsCount(String serviceName, String serviceVersion, Date invocationsDate) {
+      DailyInvocationStatistic dailyInvocationStatistic = getServiceInvocations(getHttpEndpoint(), serviceName, serviceVersion, invocationsDate);
+
+      if (dailyInvocationStatistic == null) {
+         return 0L;
+      }
+
+      BigDecimal count = dailyInvocationStatistic.getDailyCount();
+
+      return count.longValue();
+   }
+
+
+   /**
+    * Returns all data from Microcks Metrics REST API about the invocations of a given service.
+    *
+    * @param microcksContainerHttpEndpoint mandatory
+    * @param serviceName                   mandatory
+    * @param serviceVersion                mandatory
+    * @param invocationsDate               nullable
+    * @return
+    */
+   private static DailyInvocationStatistic getServiceInvocations(String microcksContainerHttpEndpoint, String serviceName, String serviceVersion, Date invocationsDate) {
+      String encodedServiceName = UrlEscapers.urlFragmentEscaper().escape(serviceName);
+      String encodedServiceVersion = UrlEscapers.urlFragmentEscaper().escape(serviceVersion);
+
+      String restApiURL = String.format("%s/api/metrics/invocations/%s/%s", microcksContainerHttpEndpoint, encodedServiceName, encodedServiceVersion);
+
+      if (invocationsDate != null) {
+         restApiURL += "?day=" + metricsApiDayFormatter.format(invocationsDate);
+      }
+
+      try {
+         Thread.sleep(100); // to avoid race condition issue when requesting Microcks Metrics REST API
```

Comment:
👍 See https://github.com/microcks/microcks/blob/master/webapp/src/main/java/io/github/microcks/web/MockControllerCommons.java#L337 => the processing of invocation event metrics is anychronous.

Reply:
Thank you for the link, interesting to see notably the event management.